Source/DafnyCore/Resolver/NameResolutionAndTypeInference/NameResolutionAndTypeInference.cs

```diff
@@ -326,6 +330,49 @@ public partial class ModuleResolver {
       }
     }
 
+    // <summary>
+    // Manually resolving NameSegments that are present in fuel attributes of reveal lemmas.
+    // This is because reveal lemmas are static and we want to allow a refence to non-static original procedures
+    // in static context in this setting.
+    //
+    // Most of the following code is copied from AnnotateRevealFunction() in OpaqueMemberRewriter.cs.
```

Comment:
Would you be able to refactor the common part out if the code is similar, so that we avoid code duplication? I haven't looked at the other side of the code.

Reply:
I looked but cannot think of a clean way to refactor the common code. Also it is not so big to warrant refactoring I think.